File: src/main/java/com/developersam/web/util/GsonUtil.java
package com.developersam.web.util;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

/**
 * All Gson related operation helper class.
 */
public final class GsonUtil {
    
    /**
     * A default gson from Google as a reference.
     */
    private static final Gson DEFAULT_GSON = new Gson();
    /**
     * A default gson for the app.
     */
    public static final Gson GSON = defaultBuild().create();
    
    /**
     * Create a {@code GsonBuilder} of default configuration.
     *
     * @return a default {@code GsonBuilder}.
     */
    private static GsonBuilder defaultBuild() {
        return new GsonBuilder().setDateFormat(DateUtil.DATE_FORMAT);
    }
    
}

File: developersam-web-core/src/main/kotlin/com/developersam/webcore/gson/GsonUtil.kt
@file:JvmName(name = "GsonUtil")

package com.developersam.webcore.gson

import com.developersam.webcore.date.commonDateFormat
import com.google.gson.Gson
import com.google.gson.GsonBuilder

/**
 * Create a [GsonBuilder] of system global configuration.
 */
private fun globalGsonBuilder(): GsonBuilder {
    return GsonBuilder().setDateFormat(commonDateFormat)
}

/**
 * A default global [Gson] for the entire app.
 */
@JvmField
val gson: Gson = globalGsonBuilder().create()